### PR TITLE
fix: use dp_world_size instead of world_size for batch_size with tensor parallelism

### PR DESCRIPTION
**Pull request by Yatimai**
# Description

Same pattern as #3444 (CP fix), applied to tensor parallelism.

`normalize_config()` multiplies `batch_size` by `world_size`, which doesn't account for TP ranks. `calculate_total_num_steps()` compensated with `* tensor_parallel_size`, masking the error.

## Motivation and Context

Follow-up to #3444, as noted by @NanoCode012:
> Fix TP calculation impacting batch size too

## How has this been tested?

4 parametrized unit tests (same pattern as the CP test in #3444):
- TP=1 on 4 GPUs: batch_size = 32
- TP=2 on 4 GPUs: batch_size = 16
- TP=4 on 4 GPUs: batch_size = 8
- TP=2 on 2 GPUs: batch_size = 8

## AI Usage Disclaimer

Yes -- minimal, diff review.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)

## Release Notes

* **Bug Fixes**
  * Improved batch size calculation when tensor parallelism is enabled
  * Refined training step estimation for distributed training scenarios

* **Tests**
  * Added test coverage for batch size calculations under tensor parallelism

### CODE REVIEW

**Review by NanoCode012 — DISMISSED** [review]
Thanks! Ci is failing from unrelated changes. I got this lint fixed in a separate PR, but if you want to run it here just to check other CI, you can do : `pre-commit run --all-files`

File: tests/test_tensor_parallel_batch_size.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+def fixture_tp_base_cfg(min_base_cfg):
+    return (
+        DictDefault(
+            micro_batch_size=2,
+            gradient_accumulation_steps=4,
+            sequence_len=2048,
+            num_epochs=1,
+        )
+        | min_base_cfg
+    )
```
[the remark this answers is not in the document]
Comment (NanoCode012) [reply tests/test_tensor_parallel_batch_size.py:21]: Incorrect

```
# dict.py line 14-15
  def __or__(self, other):
      return DictDefault(super().__ror__(other))  # Note: __ror__,
  not __or__
```

[the remark this answers is not in the document]
Comment (Yatimai) [reply tests/test_tensor_parallel_batch_size.py:21]: Fixed in the force push — merge order is now min_base_cfg | DictDefault(...).

[the remark this answers is not in the document]
Comment (NanoCode012) [reply tests/test_tensor_parallel_batch_size.py:21]: @Yatimai , this suggestion is wrong. If you run pytest locally, you can see it

**Comment by Yatimai** [comment]
You're right, sorry about that : I followed CodeRabbit's suggestion without checking. Reverted to the original merge order in the force push.

**Comment by Yatimai** [comment]
Tests fixed, mocked load_model_config to bypass the tie_word_embeddings validation. All 6 PyTest jobs pass. Lint failures are pre-existing.

**Review by NanoCode012 — APPROVED** [review]
